lualib/lua_redis.lua

```diff
@@ -879,7 +879,6 @@ local function exec_redis_script(id, params, callback, args)
     end
 
     if not args_modified then
-      table.insert(args, 1, tostring(#args))
```

Comment:
What is the reason to do it?

Reply:
The reason is we have key-arguments and non-key arguments: https://redis.io/commands/eval and we can't infer how many of each (as it is we assume they are all keyword arguments which is wrong).

Reply:
We use keyed args only as far as I know. That's why I've added this assumption. It makes this call more errors prone and more simple.

Reply:
That's not so in dmarc, ratelimit and url_tags.

Reply:
I still don't like this idea. Perhaps, we should push args somehow else? Varargs or whatever. Setting keys count explicitly is something I want to avoid at any price.

Reply:
Sure, that's a possibility; we could have table argument for keys of each type.

Reply:
Why not just change its signature to the following one:

```lua
function exec_redis_script(id, params, callback, keys, args)
  if args then ... end
end
```